firehose/client.go

```diff
@@ -62,13 +63,20 @@ func (c *Client) Start() {
 		}
 	}
 
-	subscriptionID := c.options.SubscriptionID
-	if len(subscriptionID) == 0 {
-		subscriptionID = "FirehosePlugin"
+	var errors <-chan error
+	var output <-chan *events.Envelope
+	if len(c.options.AppGUID) != 0 {
```

Comment:
I am with you on this. It reads weird. This is a pattern that we started using on the PCF Metrics team, and so I used it because of muscle memory. 

Honestly, I like the `c.options.AppGUID != ""` better. It's intention is immediately clear, but the length check to me is harder to read (I have to understand if the checked variable is a slice or a string).